aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -183,18 +285,25 @@ void printTensorNDArray(const Tensor& t) {
     }
     srcBuf = getMTLBufferStorage(_tensor);
   }
+
   // tensor.numel() could be zero, but tensor is valid as long as the buffer size is non-zero.
   // if buffer size is zero in here, it's not a user error. It could be a missing check for
   // tensor.numel() == 0 in our internal implementations of ops.
   TORCH_INTERNAL_ASSERT([srcBuf length] > 0, "Placeholder tensor is empty!");
-
   const MPSDataType mpsDataType = _tensor.dim() == 0 ? getMPSScalarType(_tensor.scalar_type()) : getMPSDataType(_tensor.scalar_type());
-  if (!mpsShape)
-    mpsShape = getMPSShape(_tensor);
 
-  _value = [[[MPSGraphTensorData alloc] initWithMTLBuffer:srcBuf
-                                                    shape:mpsShape
-                                                 dataType:mpsDataType] autorelease];
+  if (src.is_view() && src.is_contiguous() && src.storage_offset()) {
+    _value = getMPSGraphTensorDataForView(src, mpsShape, mpsDataType);
```

Comment:
Default value for mpsShape is null in this constructor. Are you sure the mpsShape is always passed and isn't null when we get here?

Reply:
Is the usage of getMPSGraphTensorDataForView() limited to Placeholders only, and not the cases where we have is_view() checks in Copy.mm?

Reply:
> Default value for mpsShape is null in this constructor. Are you sure the mpsShape is always passed and isn't null when we get here?

This case is treated at line 187 (https://github.com/kulinseth/pytorch/pull/146/files#diff-c35f3c3f2f56d77db0e1b290ea4ca49e78e016c5732d319863ebb1c88cd775bcR187):
```
 if (hasMPSShape) {
     src_ndim_view = [mpsShape count];
     src_view_shape.reserve(src_ndim_view);
     for (const auto i : c10::irange(src_ndim_view)) {
       src_view_shape[i] = [mpsShape[i] intValue];
     }
   } else {
     src_ndim_view = src.dim();
     src_view_shape = src.sizes().vec();
   }
```

Reply:
> Is the usage of getMPSGraphTensorDataForView() limited to Placeholders only, and not the cases where we have is_view() checks in Copy.mm?

Yes, we should also do it there - I'll create another PR for that change